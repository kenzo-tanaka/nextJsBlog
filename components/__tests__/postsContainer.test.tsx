import React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import PostsContainer from "../postsContainer";
import { PostData } from "@types";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
    };
  },
}));

const posts: PostData[] = [
  {
    slug: "test-slug",
    title: "Test title",
    date: "2021-04-17",
    content: "This is a test post",
    category: "work",
  },
];
const renderer = createRenderer();

test("PostsContainer", () => {
  renderer.render(<PostsContainer posts={posts} />);
  const renderedOutput = renderer.getRenderOutput();
  expect(renderedOutput).toMatchSnapshot();
});
