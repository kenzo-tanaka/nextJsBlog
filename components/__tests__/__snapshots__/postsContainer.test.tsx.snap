// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsContainer 1`] = `
<React.Fragment>
  <Profile />
  <MobileCategoryMenu />
  <hr
    className="my-5 hidden sm:block"
  />
  <div
    className="grid grid-cols-6 gap-4"
  >
    <div
      className="hidden md:block"
    >
      <CategoryMenu />
    </div>
    <div
      className="col-span-6 md:col-span-5"
    >
      <PostsList
        posts={
          Array [
            Object {
              "category": "work",
              "content": "This is a test post",
              "date": "2021-04-17",
              "slug": "test-slug",
              "title": "Test title",
            },
          ]
        }
      />
    </div>
  </div>
</React.Fragment>
`;
